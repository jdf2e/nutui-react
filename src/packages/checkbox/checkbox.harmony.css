.nut-checkbox {
  display: flex;
  align-items: center;
}
.nut-checkbox-icon {
  color: #bfbfbf;
  font-size: 16px;
}
.nut-checkbox-icon-checked {
  color: #fa2c19;
  transition-duration: 0.3s;
  transition-property: color, border-color, background-color;
}
.nut-checkbox-icon-checked.nut-checkbox-icon-disabled {
  color: #fd9d94;
}
.nut-checkbox-label {
  margin-left: 4px;
  font-size: 12px;
  color: #1a1a1a;
}
.nut-checkbox-label-disabled {
  color: #bfbfbf;
}
.nut-checkbox-icon-indeterminate {
  color: #fa2c19;
}
.nut-checkbox-icon-indeterminate.nut-checkbox-icon-disabled {
  color: #fd9d94;
}
.nut-checkbox-icon-disabled {
  color: #bfbfbf;
}
.nut-checkbox-reverse {
  flex-direction: row-reverse;
}
.nut-checkbox-reverse .nut-checkbox-label {
  margin-right: 4px;
  margin-left: 0;
}
.nut-checkbox-button {
  position: relative;
  display: inline-flex;
  align-items: center;
  padding: 5px 18px;
  font-size: 12px;
  background: #f6f6f6;
  border-radius: 15px;
  color: #1a1a1a;
  box-sizing: border-box;
  border: 1px solid #f6f6f6;
  overflow: hidden;
}
.nut-checkbox-button-active {
  background: #ffeae8;
  color: #fa2c19;
  border: 1px solid #fa2c19;
}
.nut-checkbox-button-disabled {
  color: #bfbfbf;
  border: 1px solid #f6f6f6;
}
.nut-checkbox-button-icon {
  position: absolute;
  right: 0;
  bottom: 0;
  width: 0;
  height: 0;
  border-top: 10px solid transparent;
  border-left: 10px solid transparent;
  border-bottom: 10px solid #fa2c19;
  border-right: 10px solid #fa2c19;
  display: flex;
  align-items: flex-end;
  justify-content: flex-end;
}
.nut-checkbox-button-icon-checked {
  width: 12px;
  height: 12px;
  position: absolute;
  color: #ffffff;
  top: 50%;
  left: 50%;
  transform: translate(-3px, -3px);
}
.nut-checkbox-button-icon .nut-icon {
  position: absolute;
  font-size: 12px;
  width: 12px;
  height: 12px;
}
.nut-checkbox-button-icon .nut-icon:before {
  top: auto;
  bottom: -22px;
  margin-left: 6px;
}
.nut-checkbox .nut-checkbox-button-active.nut-checkbox-button-disabled {
  background: #bfbfbf;
  color: #ffffff;
  border: 1px solid #bfbfbf;
}
.nut-checkbox-list-item {
  width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  padding: 12px 12px 12px 0;
  border-top: 1px solid rgba(0, 0, 0, 0.06);
}
.nut-checkbox-list-item .nut-checkbox-label {
  flex: auto;
}
.nut-checkbox-list-item .nut-icon {
  flex: none;
}

[dir=rtl] .nut-checkbox-label {
  margin-left: 0;
  margin-right: 4px;
}

.nut-rtl .nut-checkbox-label {
  margin-left: 0;
  margin-right: 4px;
}
[dir=rtl] .nut-checkbox-reverse .nut-checkbox-label {
  margin-left: 4px;
  margin-right: 0;
}
.nut-rtl .nut-checkbox-reverse .nut-checkbox-label {
  margin-left: 4px;
  margin-right: 0;
}
[dir=rtl] .nut-checkbox-button-icon {
  right: auto;
  left: 0;
  border-right: 10px solid transparent;
  border-left: 10px solid #fa2c19;
}
.nut-rtl .nut-checkbox-button-icon {
  right: auto;
  left: 0;
  border-right: 10px solid transparent;
  border-left: 10px solid #fa2c19;
}
[dir=rtl] .nut-checkbox-button-icon-checked,
.nut-rtl .nut-checkbox-button-icon-checked {
  left: auto;
  right: 50%;
  transform: translate(3px, -3px);
}
[dir=rtl] .nut-checkbox-button-icon .nut-icon:before,
.nut-rtl .nut-checkbox-button-icon .nut-icon:before {
  margin-left: 0;
  margin-right: 6px;
}