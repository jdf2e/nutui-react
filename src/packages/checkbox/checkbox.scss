@import '../checkboxgroup/checkboxgroup.scss';

.nut-checkbox {
  display: flex;
  align-items: center;

  &-icon {
    color: $color-text-disabled;
    font-size: $checkbox-icon-font-size;
  }

  &-icon-checked {
    color: $color-primary;
    transition-duration: 0.3s;
    transition-property: color, border-color, background-color;

    &.nut-checkbox-icon-disabled {
      color: $color-primary-disabled;
    }
  }

  &-label {
    margin-left: $checkbox-label-margin-left;
    font-size: $checkbox-label-font-size;
    color: $checkbox-label-color;

    &-disabled {
      color: $color-text-disabled;
    }
  }

  &-icon-indeterminate {
    color: $color-primary;

    &.nut-checkbox-icon-disabled {
      color: $color-primary-disabled;
    }
  }

  &-icon-disabled {
    color: $color-text-disabled;
  }

  &-reverse {
    flex-direction: row-reverse;

    .nut-checkbox-label {
      margin-right: $checkbox-label-margin-left;
      margin-left: 0;
    }
  }

  &-button {
    position: relative;
    display: inline-flex;
    align-items: center;
    padding: $checkbox-button-padding;
    font-size: $checkbox-button-font-size;
    background: $checkbox-button-background;
    border-radius: $checkbox-button-border-radius;
    color: $checkbox-label-color;
    box-sizing: border-box;
    border: 1px solid $checkbox-button-background;
    overflow: hidden;

    &-active {
      background: $color-primary-light;
      color: $color-primary;
      border: $checkbox-button-active-border;
    }

    &-disabled {
      color: $color-text-disabled;
      border: 1px solid $checkbox-button-background;
    }

    &-icon {
      position: absolute;
      right: 0;
      bottom: 0;
      width: 0;
      height: 0;
      border-top: 10px solid transparent;
      border-left: 10px solid transparent;
      border-bottom: 10px solid $color-primary;
      border-right: 10px solid $color-primary;
      display: flex;
      align-items: flex-end;
      justify-content: flex-end;

      &-checked {
        width: 12px;
        height: 12px;
        position: absolute;
        color: $white;
        top: 50%;
        left: 50%;
        transform: translate(-3px, -3px);
      }

      // background-color: $color-primary;
      .nut-icon {
        position: absolute;
        font-size: 12px;
        width: 12px;
        height: 12px;

        &:before {
          top: auto;
          bottom: -22px;
          margin-left: 6px;
        }
      }
    }
  }

  .nut-checkbox-button-active.nut-checkbox-button-disabled {
    background: $color-text-disabled;
    color: $white;
    border: 1px solid $color-text-disabled;
  }

  &-list-item {
    width: 100%;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding: $checkbox-list-item-padding;
    border-top: $checkbox-list-item-border;

    .nut-checkbox-label {
      flex: auto;
    }

    .nut-icon {
      flex: none;
    }
  }
}

[dir='rtl'] .nut-checkbox,
.nut-rtl .nut-checkbox {
  &-label {
    margin-left: 0;
    margin-right: $checkbox-label-margin-left;
  }

  &-reverse {
    .nut-checkbox-label {
      margin-left: $checkbox-label-margin-left;
      margin-right: 0;
    }
  }

  &-button {
    &-icon {
      right: auto;
      left: 0;
      border-right: 10px solid transparent;
      border-left: 10px solid $color-primary;

      &-checked {
        left: auto;
        right: 50%;
        transform: translate(3px, -3px);
      }

      .nut-icon {
        &:before {
          margin-left: 0;
          margin-right: 6px;
        }
      }
    }
  }
}
