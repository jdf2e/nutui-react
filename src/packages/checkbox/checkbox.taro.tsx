import React, {
  FunctionComponent,
  ReactNode,
  useContext,
  useEffect,
  useState,
} from 'react'
import { Checked, CheckDisabled, CheckNormal } from '@nutui/icons-react-taro'
import classNames from 'classnames'
import CheckboxGroup from '@/packages/checkboxgroup/index.taro'
import { BasicComponent, ComponentDefaults } from '@/utils/typings'
import Context from '../checkboxgroup/context'
import { usePropsValue } from '@/utils/use-props-value'
import { CheckboxLabelPosition } from '@/packages/checkboxgroup/types'

export type CheckboxShape = 'button' | 'round'

export interface CheckboxProps extends BasicComponent {
  checked: boolean
  disabled: boolean
  defaultChecked: boolean
  shape: CheckboxShape
  labelPosition: CheckboxLabelPosition
  icon: ReactNode
  activeIcon: ReactNode
  indeterminateIcon: ReactNode
  value: string | number
  indeterminate: boolean
  label: ReactNode
  onChange: (value: boolean) => void
}

const defaultProps = {
  ...ComponentDefaults,
  disabled: false,
  shape: 'round',
  labelPosition: 'right',
  icon: null,
  activeIcon: null,
  indeterminateIcon: null,
  onChange: (value) => {},
} as CheckboxProps

const classPrefix = 'nut-checkbox'
export const Checkbox: FunctionComponent<
  Partial<CheckboxProps> &
    Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'>
> & { Group: typeof CheckboxGroup } = (props) => {
  const { children } = {
    ...defaultProps,
    ...props,
  }
  const {
    icon,
    label,
    className,
    activeIcon,
    checked,
    value,
    defaultChecked,
    shape,
    disabled,
    onChange,
    indeterminate,
    indeterminateIcon,
    ...others
  } = props as any
  // eslint-disable-next-line prefer-const
  let { labelPosition, ...rest } = others
  const ctx = useContext(Context)

  let [innerChecked, setChecked] = usePropsValue<boolean>({
    value: props.checked,
    defaultValue: props.defaultChecked,
    finalValue: defaultChecked,
    onChange,
  })
  // eslint-disable-next-line prefer-const
  let [innerDisabled, setDisabled] = useState(disabled)
  const [innerIndeterminate, setIndeterminate] = useState(indeterminate)

  useEffect(() => {
    setDisabled(disabled)
  }, [disabled])

  useEffect(() => {
    setIndeterminate(indeterminate)
  }, [indeterminate])

  if (ctx) {
    if (ctx.labelPosition !== undefined) {
      labelPosition = ctx.labelPosition
    }
    innerDisabled = ctx.disabled !== undefined ? ctx.disabled : innerDisabled
    innerChecked = ctx.value.includes(value)
    setChecked = (checked: boolean) => {
      if (ctx.disabled) return
      if (checked) ctx.check(value)
      if (!checked) ctx.uncheck(value)
    }
  }

  const renderIcon = () => {
    if (innerDisabled) {
      if (innerIndeterminate) {
        return <CheckDisabled className={color()} />
      }
      if (innerChecked) {
        return <Checked className={color()} />
      }
      return <CheckDisabled className={color()} />
    }
    if (!innerChecked) {
      return React.isValidElement(icon) ? (
        icon
      ) : (
        <CheckNormal className={color()} />
      )
    }
    if (innerIndeterminate) {
      return React.isValidElement(indeterminateIcon) ? (
        indeterminateIcon
      ) : (
        <CheckDisabled className={color()} />
      )
    }
    return React.isValidElement(activeIcon) ? (
      activeIcon
    ) : (
      <Checked className={color()} />
    )
  }
  const color = () => {
    const cls = `${classPrefix}-icon `
    if (innerDisabled) {
      if (innerChecked && !innerIndeterminate) {
        return `${cls}${classPrefix}-icon-checked ${classPrefix}-icon-disabled`
      }
      return `${cls}${classPrefix}-icon-disabled`
    }
    if (innerChecked) {
      if (innerIndeterminate) {
        return `${cls}${classPrefix}-icon-indeterminate`
      }
      return `${cls}${classPrefix}-icon-checked`
    }
    return cls
  }
  const renderLabel = () => {
    return (
      <span
        className={classNames(`${classPrefix}-label `, {
          [`${classPrefix}-label-disabled`]: innerDisabled,
        })}
      >
        {children || label}
      </span>
    )
  }

  const handleClick = () => {
    // 禁用的时候直接返回
    if (disabled) return
    // 先转换状态
    const latestChecked = !innerChecked
    // 判断是不是有 context 和 max，有的话需要判断是不是超过最大限制
    if (ctx && ctx.max !== undefined) {
      if (latestChecked && ctx.value.length >= ctx.max) {
        return ctx.onLimit?.('max')
      }
    }
    setChecked(latestChecked)
  }

  const renderButton = () => {
    return (
      <div
        className={classNames(`${classPrefix}-button`, {
          [`${classPrefix}-button-active`]: innerChecked,
          [`${classPrefix}-button-disabled`]: disabled,
        })}
      >
        {children || label}
        {innerChecked && activeIcon ? (
          <div className={classNames(`${classPrefix}-button-icon`)}>
            {activeIcon}
          </div>
        ) : null}
      </div>
    )
  }

  const renderCheckboxItem = () => {
    if (shape === 'button') {
      return renderButton()
    }
    return (
      <>
        {renderIcon()}
        {renderLabel()}
      </>
    )
  }

  return (
    <div
      className={classNames(
        classPrefix,
        {
          [`${classPrefix}-reverse`]: labelPosition === 'left',
        },
        className
      )}
      {...rest}
      onClick={handleClick}
    >
      {renderCheckboxItem()}
    </div>
  )
}

Checkbox.defaultProps = defaultProps
Checkbox.displayName = 'NutCheckBox'
Checkbox.Group = CheckboxGroup
