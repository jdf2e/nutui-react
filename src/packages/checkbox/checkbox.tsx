import React, {
  FunctionComponent,
  useContext,
  useEffect,
  useState,
} from 'react'
import { Checked, CheckDisabled, CheckNormal } from '@nutui/icons-react'
import CheckboxGroup from '@/packages/checkboxgroup'
import bem from '@/utils/bem'
import { BasicComponent, ComponentDefaults } from '@/utils/typings'
import Context from '../checkboxgroup/context'

export interface CheckboxProps extends BasicComponent {
  checked: boolean
  disabled: boolean
  textPosition: 'left' | 'right'
  iconSize: string | number
  icon: React.ReactNode
  checkedIcon: React.ReactNode
  indeterminateIcon: React.ReactNode
  iconClassPrefix: string
  iconFontClassName: string
  indeterminate: boolean
  label: string | number
  onChange: (state: boolean, label: string) => void
}

const defaultProps = {
  ...ComponentDefaults,
  checked: false,
  disabled: false,
  textPosition: 'right',
  iconSize: 18,
  icon: 'check-normal',
  checkedIcon: 'checked',
  iconClassPrefix: 'nut-icon',
  iconFontClassName: 'nutui-iconfont',
  indeterminateIcon: 'check-disabled',
  onChange: (state, label) => {},
} as CheckboxProps
export const Checkbox: FunctionComponent<
  Partial<CheckboxProps> &
    Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'>
> & { Group: typeof CheckboxGroup } = (props) => {
  const { children } = {
    ...defaultProps,
    ...props,
  }
  const b = bem('checkbox')
  const {
    icon,
    iconSize,
    label,
    className,
    checkedIcon,
    checked,
    disabled,
    onChange,
    indeterminate,
    iconClassPrefix,
    iconFontClassName,
    indeterminateIcon,
    ...others
  } = props as any
  // eslint-disable-next-line prefer-const
  let { textPosition, ...rest } = others
  const ctx = useContext(Context)

  let [innerChecked, setInnerChecked] = useState(checked)
  // eslint-disable-next-line prefer-const
  let [innerDisabled, setDisabled] = useState(disabled)
  const [_indeterminate, setIndeterminate] = useState(indeterminate)

  useEffect(() => {
    !ctx && setInnerChecked(checked)
    setDisabled(disabled)
    setIndeterminate(indeterminate)
  }, [disabled, checked, indeterminate])

  if (ctx) {
    if (ctx.textPosition !== undefined) {
      textPosition = ctx.textPosition
    }
    innerDisabled = ctx.disabled
    innerChecked = ctx.checkedValue.includes(label)
    setInnerChecked = (checked: boolean) => {
      if (ctx.disabled) return
      if (checked) ctx.check(label)
      if (!checked) ctx.uncheck(label)
    }
  }

  const renderIcon = () => {
    if (!innerChecked) {
      return React.isValidElement(icon) ? (
        icon
      ) : (
        <CheckNormal width={iconSize} height={iconSize} className={color()} />
      )
    }
    if (_indeterminate) {
      return React.isValidElement(indeterminateIcon) ? (
        indeterminateIcon
      ) : (
        <CheckDisabled width={iconSize} height={iconSize} className={color()} />
      )
    }
    return React.isValidElement(checkedIcon) ? (
      checkedIcon
    ) : (
      <Checked width={iconSize} height={iconSize} className={color()} />
    )
  }
  const color = () => {
    if (innerDisabled) {
      return 'nut-checkbox__icon--disable'
    }
    if (innerChecked) {
      if (_indeterminate) {
        return 'nut-checkbox__icon--indeterminate'
      }
      return 'nut-checkbox__icon'
    }
    return 'nut-checkbox__icon--unchecked'
  }
  const renderLabel = () => {
    return (
      <span className={`${b('label', { disabled: innerDisabled })} `}>
        {children || label}
      </span>
    )
  }

  const handleClick = () => {
    // 禁用的时候直接返回
    if (disabled) return
    // 先转换状态
    const latestChecked = !innerChecked
    // 判断是不是有 context 和 max，有的话需要判断是不是超过最大限制
    if (ctx && ctx.max !== undefined) {
      if (latestChecked && ctx.checkedValue.length >= ctx.max) return
    }
    onChange && onChange(latestChecked, label || (children as string))
    setInnerChecked(latestChecked)
  }

  return (
    <div
      className={`${b({ reverse: textPosition === 'left' })} ${
        className || ''
      }`}
      {...rest}
      onClick={handleClick}
    >
      {renderIcon()}
      {renderLabel()}
    </div>
  )
}

Checkbox.defaultProps = defaultProps
Checkbox.displayName = 'NutCheckBox'
Checkbox.Group = CheckboxGroup
