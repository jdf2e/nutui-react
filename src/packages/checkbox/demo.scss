.demo-check {
  margin-right: 10px;
}
.nut-checkboxgroup {
  display: flex;
  flex-wrap: wrap;
}
.nut-checkbox {
  display: flex;
  margin-right: 20px;
  .nut-checkbox__label {
    margin-left: 10px;
  }
}
