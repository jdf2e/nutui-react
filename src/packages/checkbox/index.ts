import { Checkbox } from './checkbox'

export type { CheckboxProps } from './checkbox'
export default Checkbox
