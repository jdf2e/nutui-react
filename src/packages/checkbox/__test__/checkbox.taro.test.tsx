import TestUtils from '@tarojs/test-utils-react'
import '@testing-library/jest-dom'
import { Checkbox } from '../checkbox.taro'

const testUtils = new TestUtils()
const execFn = (desc: string, component: any, props: object, cb: any) => {
  it(desc, async () => {
    await testUtils.mount(component, {
      props,
    })
    cb()
    testUtils.unmout()
  })
}
describe('checkbox-taro', () => {
  execFn(
    'should match snapshot',
    Checkbox,
    {
      labelPosition: 'left',
      label: '复选框',
      checked: true,
    },
    () => {
      expect(testUtils.html()).toMatchSnapshot()
    }
  )

  execFn(
    'should props correctly',
    Checkbox,
    {
      style: { color: 'red' },
      className: 'test',
      labelPosition: 'left',
      label: '复选框',
      disabled: true,
    },
    () => {
      expect(
        testUtils.queries.querySelector('.nut-checkbox-label-disabled')
      ).toBeInTheDocument()
      expect(testUtils.queries.querySelector('.nut-checkbox')).toHaveClass(
        'test'
      )
      expect(testUtils.queries.querySelector('.nut-checkbox')).toHaveStyle(
        'color: red'
      )
      expect(testUtils.queries.queryByText('复选框')).toBeInTheDocument()
    }
  )

  execFn(
    'Render checkboxs by configure indeterminate',
    Checkbox,
    {
      value: '1',
      checked: true,
      label: 'labe1',
      indeterminate: true,
    },
    () => {
      expect(
        testUtils.queries.querySelector('.nut-checkbox-icon-indeterminate')
      ).toBeTruthy()
    }
  )

  execFn(
    'Render checkboxs by configure disabled',
    Checkbox,
    {
      value: '1',
      checked: true,
      label: 'labe1',
      disabled: true,
    },
    () => {
      expect(
        testUtils.queries.querySelector('.nut-checkbox-icon-disabled')
      ).toBeTruthy()
    }
  )

  execFn(
    'Render checkboxs by configure disabled and indeterminate',
    Checkbox,
    {
      value: '1',
      checked: true,
      label: 'labe1',
      disabled: true,
      indeterminate: true,
    },
    () => {
      expect(
        testUtils.queries.querySelector('.nut-checkbox-icon-indeterminate')
      ).toBeTruthy()
      expect(
        testUtils.queries.querySelector('.nut-checkbox-icon-indeterminate')
      ).toHaveClass('nut-checkbox-icon-disabled')
    }
  )
})
