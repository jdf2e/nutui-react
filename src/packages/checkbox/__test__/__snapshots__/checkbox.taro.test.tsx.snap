// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox-taro should match snapshot 1`] = `
"[36m<body>[39m
  [36m<div>[39m
    [36m<taro-view-core[39m
      [33mclass[39m=[32m"nut-checkbox nut-checkbox-reverse"[39m
    [36m>[39m
      [36m<i[39m
        [33mclass[39m=[32m" nut-icon nut-icon-Checked nut-checkbox-icon nut-checkbox-icon-checked"[39m
        [33mstyle[39m=[32m"background-color: currentColor; mask: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMDI0IDEwMjQiPjxwYXRoIGZpbGw9IiMzMzMiIGQ9Ik01MTIgMEMyMzAuNCAwIDAgMjMwLjQgMCA1MTJzMjMwLjQgNTEyIDUxMiA1MTIgNTEyLTIzMC40IDUxMi01MTJTNzkzLjYgMCA1MTIgMG0yODEuNiAzNjkuNzhjLTE0LjIyIDExLjM4LTE5My40MiAxMzAuODQtMzIxLjQyIDMyMS40MiAwIDAgMCAyLjg0LTIuODUgMi44NC04LjUzIDUuNjktNDguMzUgMzYuOTgtODguMTctOC41My0zOS44Mi01MS4yLTYyLjU4LTk5LjU2LTE0Mi4yMy0xNDIuMjItMi44NCAwLTIuODQtMi44NS0yLjg0LTIuODUtOC41My0xMS4zOC0zOS44Mi01Ni44OSAxOS45MS01Ni44OCA0NS41MSAwIDkxLjAyIDExLjM4IDE2Mi4xMyA3My45NSA1LjY5IDUuNjkgMTQuMjIgNS42OSAxNy4wNyAwIDM0LjEzLTM5LjgyIDE3My41MS0xOTAuNTggMzMyLjgtMjM4LjkzIDAgMCAxOS45MS0yLjg1IDMxLjI5IDE0LjIyIDUuNjkgMTEuMzggMTEuMzggMjIuNzYtNS42OSAzNi45OCIvPjwvc3ZnPg==')  0 0/100% 100% no-repeat;"[39m
      [36m/>[39m
      [36m<span[39m
        [33mclass[39m=[32m"nut-checkbox-label "[39m
      [36m>[39m
        [0mÂ§çÈÄâÊ°Ü[0m
      [36m</span>[39m
    [36m</taro-view-core>[39m
  [36m</div>[39m
[36m</body>[39m"
`;
